src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,207 @@
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { useEffect } from 'react';
+import { Card, Element } from '@invoiceninja/cards';
+import { InputField, SelectField } from '@invoiceninja/forms';
+import {
+  defaultTransactionProperties,
+  transactionTypes,
+} from 'common/constants/transactions';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { useTitle } from 'common/hooks/useTitle';
+import { endpoint } from 'common/helpers';
+import { TransactionInput } from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useTransactionPages } from '../common/hooks/useTransactionPages';
+import { useTransactionValidation } from '../common/hooks/useTransactionValidation';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import toast from 'react-hot-toast';
+import { useNavigate, useParams } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+
+const Edit = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const { id } = useParams<string>();
+  const currencies = useCurrencies();
+  const pages = useTransactionPages();
+  const { documentTitle } = useTitle('edit_transaction');
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errors, setErrors] = useState<TransactionValidation>();
+  const [transaction, setTransaction] = useState<TransactionInput>(
+    defaultTransactionProperties
+  );
+  const { checkValidation, setValidation } = useTransactionValidation();
```

Comment:
Please add blank lines to improve clarity.

Reply:
> Please add blank lines to improve clarity.

done